### PR TITLE
[notifications][android]: don't interpret custom sounds without vibrations as silent

### PR DESCRIPTION
**Pull request by pennersr**
# Why

See:

https://github.com/expo/expo/issues/20351#issuecomment-2505988515

You will end up in the `setSilent(true)` branch when you have a notification with a custom sound. Because, in that case, `shouldPlayDefaultSound` will be `false`, and the original logic will incorrectly conclude:

> Notification will not vibrate or play sound, regardless of channel

# How

Needed to fix the (incorrectly closed) issue #20351 

# Test Plan

I tested this by sending actual FCM notifications. Other than that, this is a one-liner for code that is clearly wrong. A human review should be able to confirm & verify.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Review by vonovak — COMMENTED** [review]
Hello and thank you for the PR!

As I see it, there are a bit too many conditions around sounds and maybe the sound handling can be simplified a bit.

Would you be willing to do at least one of:

- (1) write a test case for the issue you're seeing [here](https://github.com/expo/expo/blob/main/packages/expo-notifications/android/src/androidTest/kotlin/expo/modules/notifications/ExpoNotificationBuilderTest.kt) 
- (2) refactor the sound-handling part?

?


Thank you!

**Comment by pennersr** [comment]
@vonovak Getting down to the root cause of this issue impacted my planning quite a bit, so I am unable to spend more at this time...

**Comment by vonovak** [comment]
> @vonovak Getting down to the root cause of this issue impacted my planning quite a bit, so I am unable to spend more at this time...

understood, can you please take the changelog suggestion from https://github.com/expo/expo/pull/33311#issuecomment-2506163735 (expand the comment) and add it to the package's changelog?

**Comment by pennersr** [comment]
@vonovak Done.

**Review by douglowder — APPROVED** [review]

**Review by vonovak — APPROVED** [review]

File: packages/expo-notifications/android/src/androidTest/kotlin/expo/modules/notifications/notifications/model/ExpoNotificationBuilderTest.kt [hunk 1]
```diff
@@ -44,11 +40,21 @@ class ExpoNotificationBuilderTest {
...
     val exNotification =
-      expo.modules.notifications.notifications.model.Notification(notificationRequest)
+      Notification(notificationRequest)
 
     return ExpoNotificationBuilder(context, exNotification, categoriesStore)
   }
 
+  private fun assertIsSilent(notification: Notification) {
+    // Verifying sound is a bit tricky because of channels; so we use this "proxy"
+    assertTrue("Notification should not be silent", notification.group == "silent")
```
Comment (douglowder) [inline packages/expo-notifications/android/src/androidTest/kotlin/expo/modules/notifications/notifications/model/ExpoNotificationBuilderTest.kt:50]:
```suggestion
    assertTrue("Notification should be silent", notification.group == "silent")
```

**Review by douglowder — APPROVED** [review]
Looks good, left one small suggestion

**Comment by vonovak** [comment]
The test failure is unrelated, so I merged. Thank you!

**Pull request merged**